pkg/services/object/get/util.go

```diff
@@ -296,3 +262,31 @@ func (h *hasherWrapper) WriteChunk(p []byte) error {
 func prettyRange(rng *object.Range) string {
 	return fmt.Sprintf("[%d:%d]", rng.GetOffset(), rng.GetLength())
 }
+
+// DirectChildWriter streams child object payload directly into destination ChunkWriter
+// while capturing the header.
+type DirectChildWriter struct {
+	obj       *object.Object
+	dest      ChunkWriter
+	headerSet bool
+}
+
+func NewDirectChildWriter(dest ChunkWriter) *DirectChildWriter {
+	return &DirectChildWriter{
+		obj:  object.New(),
```

Comment:
why we init it? Seems like we can drop this, and then drop `HeaderSet` cuz it will be the same as `w.hdr != nil`